operators/src/engine/result_descriptor.rs

```diff
@@ -95,22 +95,37 @@ impl RasterResultDescriptor {
 pub struct VectorResultDescriptor {
     pub data_type: VectorDataType,
     pub spatial_reference: SpatialReferenceOption,
-    pub columns: HashMap<String, FeatureDataType>,
+    pub columns: HashMap<String, VectorColumnInfo>,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct VectorColumnInfo {
+    pub data_type: FeatureDataType,
+    pub measurement: Measurement,
 }
 
 impl VectorResultDescriptor {
     /// Create a new `VectorResultDescriptor` by only modifying the columns
     #[must_use]
     pub fn map_columns<F>(&self, f: F) -> Self
     where
-        F: Fn(&HashMap<String, FeatureDataType>) -> HashMap<String, FeatureDataType>,
+        F: Fn(&HashMap<String, VectorColumnInfo>) -> HashMap<String, VectorColumnInfo>,
     {
         Self {
             data_type: self.data_type,
             spatial_reference: self.spatial_reference,
             columns: f(&self.columns),
         }
     }
+
+    pub fn column_data_type(&self, column: &str) -> Option<FeatureDataType> {
```

Comment:
shouldn't this return a `Result` and give an columnNotFound error?

Reply:
But a HashMap also gives an Option back and not an error that states: not found.
So… there is only one meaning currently :smile:

Reply:
yeah that's exactly the reason why we do not use hashmaps directly but wrap them and handle the cases where something is not where it is supposed t obe.

Reply:
But this is not a user API. You have to wrap it into an error anyway at your, for instance, operator. This is a convenience function for accessing the hash map with more than one value.